compiler/passes/convert-typed-uast.cpp

```diff
@@ -5201,10 +5254,9 @@ bool TConverter::enter(const Select* node, RV& rv) {
       } else if (!allParamFalse) {
         auto cond = getWhenCond(this, rv, when, selectSym);
 
-        auto thenBlock = new BlockStmt();
-        pushBlock(thenBlock);
+        // Traversing the body creates a BlockStmt we can remove
         when->body()->traverse(rv);
```

Comment:
Maybe we can add a little widget `convertBlock` that does this pattern in case it pops up again in the future.